### ISSUE TITLE
 imports to variable with same name in constructor

### ISSUE THREAD

**Issue by runeabrahams1** [sweepline/eslint-plugin-unused-imports#4]
If you name a variable the same as an import in typescript, like this:
```ts
import {Service} from "..."
constructor(private Service: Service) {}
```

The plugin will report Service as  an unused  import.
If you run --fix the import is removed, which is wrong.

**Comment by sweepline** [comment]
I cannot reproduce this, I am testing with 

```ts
import { Service } from "...";

class Whoop {
	foo: Service;
	constructor(private Service: Service) {
		this.foo = Service;
	}
}
```

**Comment by runeabrahams1** [comment]
Remove foo:Service. That should give the error

Mvh
Rune Abrahamsen

ons. 12. feb. 2020, 10:15 skrev Mikkel Holmer <notifications@github.com<mailto:notifications@github.com>>:

I cannot reproduce this, I am testing with

import { Service } from "...";

class Whoop {
        foo: Service;
        constructor(private Service: Service) {
                this.foo = Service;
        }
}

**Comment by runeabrahams1** [comment]
Import { Service } from "... " 
class Whoop {
	constructor(private Service: Service) {
		this.Service.doSomething();
	}
}

Think this should reproduce it, but I'll try in our code if it don't

**Comment by developer239** [comment]
@runeabrahams1 I was worried at first but I couldn't reproduce this issue. I have example Nest.js application here (it looks like you are using either Nest.js or Angular)

https://github.com/developer239/nest-rest-api

You can see me using the plugin here: https://github.com/code-quality-resources/eslint-config-base
and here: https://github.com/code-quality-resources/eslint-config-typescript

I hope this helps you debugging and that the plugin actually works as it is supposed to. So far it look amazing. 😄 💪

**Comment by sweepline** [comment]
I took another look and it's definitely an upstream issue, closing..

**Comment by lelonek1** [comment]
For others who have this problem, the upstream meta-issue tracking the changes needed to fix this is https://github.com/typescript-eslint/typescript-eslint/issues/1856.
